### PR TITLE
Remove Modernizr, keep html5shiv (Webpack edition)

### LINKED ISSUE #724: Screen flashes on load
### Problem

When I go to any page using the RTD theme, my browser noticeably flashes white for a few dozen milliseconds before the page shows up. When I comment out `modernizr.min.js` from my HTML pages, the white flashes go away.

### Reproducible Project

I see this in every project that uses the RTD theme. Which means I can repro it here too: https://sphinx-rtd-theme.readthedocs.io/en/stable

Steps to repro:

1) Go to https://sphinx-rtd-theme.readthedocs.io/en/stable
(This already repros the bug, but it's not so easy to notice on first load. So let's go to step 2)

2) Click on "Configuration" on the left sidebar.
Expected: the Configuration page seamlessly loads
Actual: the screen flashes white _and then_ the Configuration page loads

3) You can repro this 100% of the time by clicking on **top level** links on the sidebar.

#### Error Logs/Results

There are no errors on the console.

### Environment Info

- Python Version: n/a
- Sphinx Version: 1.8.4
- RTD Theme Version: 0.4.3
- Browser: Chrome Version 72.0.3626.96 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by jessetan**
Modernizr has been part of the theme for a long time, although it is not used. Since its presence has downsides (reflow on load, increased theme size), we want to remove it.
To keep backward compatibility with older IEs, I've kept html5shiv in this PR so there is less chance of breakage.

This is a rework of #756 after our move to Webpack and friends. Rebasing did not make much sense.
Fixes #724, #525. Related #414. Supersedes #624 and #756

### CODE REVIEW

File: sphinx_rtd_theme/layout.html [hunk 1]
```diff
@@ -9,8 +9,7 @@
 {%- set lang_attr = 'en' if language == None else (language | replace('_', '-')) %}
 
 <!DOCTYPE html>
-<!--[if IE 8]><html class="no-js lt-ie9" lang="{{ lang_attr }}" > <![endif]-->
-<!--[if gt IE 8]><!--> <html class="no-js" lang="{{ lang_attr }}" > <!--<![endif]-->
+<html lang="{{ lang_attr }}" >
```
Comment (agjohnson) [inline sphinx_rtd_theme/layout.html]: Was modernizer also responsible for the `no-js` class here? I was considering using that in the future for no-JS compatibility on some of the CSS, but that functionality would be easy enough to replicate if so.

Comment (jessetan) [reply sphinx_rtd_theme/layout.html]: I think the `no-js` was there by default and then removed by Modernizr.
We can do something similar if we want to support users without JS.

**Comment by jessetan** [comment]
Any particular IE versions you want to see? Microsoft no longer offers free screenshots using Browserstack, that requires a paid account.
They do offer VMs with IE8 and up.

**Comment by agjohnson** [comment]
I tried saucelabs, but they dropped IE8 support. IE9 looked as expected though. I tried a demo account on browserstack, but couldn't test IE8. I even tried ie8 through wine to no avail.

I guess I'm not sure how thorough we need to be here then, testing sounds annoying. IE8 only accounts for 0.1% of RTD traffic, which also isn't 100% for projects using this theme.

**Comment by jessetan** [comment]
Perhaps the best way to get feedback on this is to release a dev version of the theme and to wait for user responses or bug reports.

**Comment by agjohnson** [comment]
Sounds good.

**Pull request merged**